cmd/sso-proxy/main.go

```diff
@@ -58,5 +59,10 @@ func main() {
 		Handler:      loggingHandler,
 	}
 
-	logger.Fatal(s.ListenAndServe())
+	// How long we will wait to allow in-flight requests to finish
+	shutdownTimeout := 30 * time.Second
```

Comment:
For sso-auth above, there's a natural shutdown timeout to use based on the request timeout we already configure at the server level.  For sso-proxy, I'm not sure there's any existing upper-limit on request time that we can use here, so I picked a fairly arbitrary number.

(30 seconds matches the time that we at BuzzFeed have configured the AWS ecs agent to wait before forcibly killing a container when it does not exit after receiving `SIGTERM`.)

We could also make this an explicit config value w/ a reasonable default, if you'd prefer.  Or, please let me know if there is some existing config value on which it would make sense to piggyback here!